### ISSUE TITLE
Changing the price of an item when receiving it

### ISSUE THREAD

**Issue by marteserede** [jekkos/opensourcepos#27]
Often my prices change for some stuff I buy.
Therefore it would be cool to be able to change the price of a product when receiving it.
If I still have items in stock, an average price shall be calculated.

**Comment by flodef** [comment]
I've got the same situation here.
For now, I'm just updating the other prices and the customer is paying the
difference.
But your solution would be good as well.

**Comment by jekkos** [comment]
There is always the possibility to adapt the item cost for each receiving. But in this case you'd want to change the unit price permanently?

Also I see in the commit that you make calculations without any predefined precision. This can lead to rounding errors when using floats with different scales. It might be better to use bc-style operations here as seen in the sale and receiving libs.

**Comment by marteserede** [comment]
Yes, I want to change the unit price permanently.
Because if not, changing the item cost would not make much sense nor have much effect: only on the receipt for the receiving would be the different item cost written, isnt it?

I changed the calculations to bc-style.

**Comment by jekkos** [comment]
I will probably want to add a config option to enable/disable this feature, as most people will expect the application to continue working as before.

**Comment by marteserede** [comment]
I added the option to enable/disable this, see the commit.
Is that the right way to do it?
I noticed that it would probably make sense to add small tooltip for the different config settings. I tried to give it a short title so it does not break in two lines, but on the other hand the title now is not really self-explaining thus a tooltip would make sense...
I also added the language-string to the csv-file, but did not generate all language files again because then my new indo-translations would be missing, i think.
